File: tests/fixtures/effects/scala/Quotes.scala
import sttp.client3._

object Quotes {
  def fetch(url: String): Option[String] = {
    val backend = HttpClientSyncBackend()
    for (attempt <- 0 until 3) {
      try {
        return Some(basicRequest.get(uri"$url").send(backend).body.toString)
      } catch {
        case e: Exception => ()
      }
    }
    None
  }
}

File: tests/fixtures/effects/php/src/Quotes.php
<?php
namespace App;

use GuzzleHttp\Client;
use GuzzleHttp\Exception\RequestException;

class Quotes
{
    private Client $client;

    public function fetch(string $url)
    {
        for ($i = 0; $i < 3; $i++) {
            try {
                return $this->client->get($url);
            } catch (RequestException $e) {
                continue;
            }
        }
        return null;
    }
}
